src/librustc_middle/mir/mod.rs

```diff
@@ -2069,6 +2070,11 @@ pub enum Rvalue<'tcx> {
     /// &x or &mut x
     Ref(Region<'tcx>, BorrowKind, Place<'tcx>),
 
+    /// Accessing a thread local static. This is inherently a runtime operation, even if llvm
+    /// treats it as an access to a static. This `Rvalue` yields a reference to the thread local
+    /// static.
+    Tls(DefId),
```

Comment:
`ThreadLocalRef` might be a better name.

Reply:
@oli-obk looks like the naming bikeshed is still open.